### ISSUE TITLE
Gloablly define Environment Variables to be passed to remote nodes.

### ISSUE THREAD

**Issue by grisuthedragon** [open-mpi/ompi#4877]
## Background information
Improve the usabilty of the OpenMPI for entry level users.  Especially Handling environment variables.

### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)
v2.0.4

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
configured with --enable-shared --enable-mpi-cxx --enable-mpi-fortran --with-sge --with-slurm --enable-mpi-threads --with-psm2 --with-pmi  LDFLAGS="-Wc,-static-intel ", Intel 2018 Compilers

(Slurm and SGE not used at the moment)

### Please describe the system on which you are running

* Operating system/version: CentOS7
* Computer hardware: Xeon Skylake Silver 4110
* Network type: OmniPATH
* Remote Execution: Password-less SSH
* All software/homes/etc on shared Filesystem.
-----------------------------

## Details of the problem
The problem comes from the fact the our applications using MKL and on the new Skylake Xeon Silver editions I have to restrict the MKL library to use only AVX2 instead of AVX512 instructions. This is controlled using an environment variable. At the moment I run the code via 
    
    module load apps/sim2
    mpirun -x MKL_ENABLE_INSTRUCTIONS -x PATH -x LD_LIBRARY_PATH -machinefile nodefile -np 64 --map-by node 

to all necessary environment variables to the compute nodes. But I find this procedure a bit complicated to explain the users although the module is loaded that they have to specify each enviroment variable that may be necessary on the other nodes as well. (Mostly my users even do not know what environment variables are) 

So is there a way to specify a list of environment variables globally for all users or copy the whole environment automatically to all other compute nodes.  Because without them, even shared objects are not found and this make OpenMPI nearly not usable for non advanced users.

**Comment by ggouaillardet** [comment]
Note the `v2.0` series is no more supported, and you should upgrade to the `v2.1` series (and I encourage you to upgrade to Open MPI `3.0.1` that will be released shortly).

Since Open MPI is available in the same directory on all the nodes, I strongly encourage you to
```
configure --enable-mpirun-prefix-by-default
```
As far as Open MPI is concerned, you won't need to export `PATH` nor `LD_LIBRARY_PATH` via `mpirun`.

Assuming you (basically)
```
export MKL_ENABLE_INSTRUCTIONS=value
```
in your module, then an other option is to
```
mpirun --mca mca_base_env_list MKL_ENABLE_INSTRUCTIONS=value ...
```
an alternative is to do that via the environment
```
export OMPI_MCA_mca_base_env_list=MKL_ENABLE_INSTRUCTIONS=value
```
and that can obviously be done in your module file.

You might still need to export `LD_LIBRARY_PATH` for `MKL`, and I can suggest you teach your endusers to use `rpath` when building their apps, so that will not be needed any more.

**Comment by melvynpan** [comment]
you can wirte env variables in ~/.bashrc if your os is centos

**Comment by rhc54** [comment]
This is fixed in master and v5.0 - there are multiple ways to export envars.

**Comment by mahendrapaipuri** [comment]
@ggouaillardet I have faced exactly same issue recently and I am exporting `LD_LIBRARY_PATH` for `MKL` using `mca_base_env_list` MCA parameter. You mentioned about using `rpath` to avoid these situations. Could you please tell me how can I do it? I tried to search online but could not find any clear info. I am building `MKL` libraries using Spack and using `lmod` to load modules. Thanks in advance.
